test/Crystallization.cpp

```diff
@@ -0,0 +1,177 @@
+{
+	// read json model setup file
+	const std::string& modelFileRelPath = std::string("/data/configuration_PBM_CSTR_growthSizeDep_benchmark1.json");
+	const std::string setupFile = std::string(getTestDirectory()) + modelFileRelPath;
+	cadet::JsonParameterProvider pp_setup(cadet::JsonParameterProvider::fromFile(setupFile));
+
+	pp_setup.pushScope("model");
+	pp_setup.pushScope("unit_001");
+
+	cadet::test::column::testJacobianAD(pp_setup, std::numeric_limits<float>::epsilon() * 100.0);
+}
+
+// Note: two entries fail, given in (row,col): (204, 190), (204, 191)
+TEST_CASE("Crystallization Jacobian verification for a CSTR with primary nucleation and growth", "[CrysToFix3]") // "[Crystallization],[UnitOp],[Jacobian]")
+{
+	// read json model setup file
+	const std::string& modelFileRelPath = std::string("/data/configuration_PBM_CSTR_primaryNucleationAndGrowth_benchmark1.json");
+	const std::string setupFile = std::string(getTestDirectory()) + modelFileRelPath;
+	cadet::JsonParameterProvider pp_setup(cadet::JsonParameterProvider::fromFile(setupFile));
+
+	pp_setup.pushScope("model");
+	pp_setup.pushScope("unit_001");
+
+	cadet::test::column::testJacobianAD(pp_setup, std::numeric_limits<float>::epsilon() * 100.0);
+}
+
+// Some more comparisons fail
+TEST_CASE("Crystallization Jacobian verification for a CSTR with primary nucleation, growth and growth rate dispersion", "[CrysToFix4]") // "[Crystallization],[UnitOp],[Jacobian]")
+{
+	// read json model setup file
+	const std::string& modelFileRelPath = std::string("/data/configuration_PBM_CSTR_primaryNucleationGrowthGrowthRateDispersion_benchmark1.json");
+	const std::string setupFile = std::string(getTestDirectory()) + modelFileRelPath;
+	cadet::JsonParameterProvider pp_setup(cadet::JsonParameterProvider::fromFile(setupFile));
+
+	pp_setup.pushScope("model");
+	pp_setup.pushScope("unit_001");
+
+	cadet::test::column::testJacobianAD(pp_setup, std::numeric_limits<float>::epsilon() * 100.0);
+}
+
+// Note: two entries fail, given in (row,col): (204, 190), (204, 191)
+TEST_CASE("Crystallization Jacobian verification for a CSTR with primary and secondary nucleation and growth", "[CrysToFix5]") // "[Crystallization],[UnitOp],[Jacobian]")
+{
+	// read json model setup file
+	const std::string& modelFileRelPath = std::string("/data/configuration_PBM_CSTR_primarySecondaryNucleationAndGrowth_benchmark1.json");
+	const std::string setupFile = std::string(getTestDirectory()) + modelFileRelPath;
+	cadet::JsonParameterProvider pp_setup(cadet::JsonParameterProvider::fromFile(setupFile));
+
+	pp_setup.pushScope("model");
+	pp_setup.pushScope("unit_001");
+
+	cadet::test::column::testJacobianAD(pp_setup, std::numeric_limits<float>::epsilon() * 100.0);
+}
+
+// FD pattern tests fail due to non-equal entries
+// AD tests fail: access violation
+TEST_CASE("Crystallization Jacobian verification for a DPFR/LRM with primary and secondary nucleation and growth", "[CrysToFix6]") // "[Crystallization],[UnitOp],[Jacobian]")
```

Comment:
@WFlynnZ this test fails with access violation error and needs some debugging, I can help you with that